test/fly.js

```diff
@@ -1,307 +1,349 @@
-    process.chdir(join(__dirname, "fixtures"))
-    const fly = new Fly()
-    yield fly
-      .source("*.txt")
-      .filter((data) => data.toString().toUpperCase())
-      .target(".")
-
-    yield fly
-      .source("*.txt")
-      .filter((data) => {
-        t.ok(data.toString() === "FOO BAR\n", "resolve source, filters and writers")
-        return data.toString().toLowerCase()
-      })
-      .target(".")
-  })
-})
+var fs = require('fs');
+var co = require('co');
+// var { readFile as read } = require('mz/fs');
+var test = require('tape').test;
+var touch = require('touch');
+var Fly = require('../lib/fly');
+var join = require('path').join;
+
+var fixtures = join(process.cwd(), 'test', 'fixtures');
+var flyfile = join(fixtures, 'flyfile.js');
+
+test('✈  fly', function (t) {
+	t.ok(Fly !== undefined, 'is defined');
+
+	var methods = [
+		'source', 'filter', 'watch', 'unwrap', 'exec',
+		'start', 'write', 'clear', 'concat', 'target', 'emit'
+	];
+
+	methods.forEach(function (method) {
+		t.ok(method !== undefined, method + ' is defined');
+	});
+
+	t.end();
+});
+
+test('✈  fly.constructor', function (t) {
+	var fly = new Fly({
+		file: flyfile,
+		host: {
+			task: function () {
+				t.equal(fly, this, 'bind tasks to fly instance');
+			}
+		}
+	});
+
+	t.ok(fly !== undefined, 'create fly instance');
+
+	t.ok(fly.tasks.task !== undefined, 'load task from host');
+	fly.tasks.task();
+
+	t.deepEqual(fly.plugins, [], 'no default plugins');
+
+	t.equal(process.cwd(), fixtures, 'switch current working directory');
+
+	t.equal(fly.file, flyfile, 'set file to path specified in the constructor');
+
+	t.end();
+});
+
+test('✈  fly.source', function (t) {
+	var fly = new Fly();
+	fly.source([[[[['*.a', ['*.b']]], ['*.c']]]]);
+	t.deepEqual(fly._.globs, ['*.a', '*.b', '*.c'], 'flatten globs');
+	t.deepEqual(fly._.filters, [], 'init empty filter list');
+	t.end();
+});
+
+test('✈  fly.filter', function (t) {
+	var fly = new Fly();
+
+	fly.filter(function (src) {
+		return src.toLowerCase();
+	});
+
+	t.equal(fly._.filters.length, 1, 'add filter to filter collection');
+
+	t.equal(fly._.filters[0].cb('FLY'), 'fly', 'add transform cb for anonymous filters');
+
+	fly.filter('myFilter', function (data) {
+		return {
+			code: data.toString().toUpperCase(),
+			ext: '.foo'
+		};
+	});
+
+	t.ok(fly.myFilter instanceof Function, 'add transform cb to fly instance for named filters');
+
+	fly.myFilter({secret: 42});
+
+	t.equal(fly._.filters[1].cb('fly').code, 'FLY', 'create transform cb function for named filter');
+	t.equal(fly._.filters[1].cb('fly').ext, '.foo', 'read extension from filter');
+	t.equal(fly._.filters[1].options.secret, 42, 'read options from filter');
+
+	try {
+		fly.filter('myFilter');
+	} catch (e) {
+		t.ok(true, 'throw an error if filter already exists');
+	}
+
+	fly.source('');
+	t.deepEqual(fly._.filters, [], 'reset filter each time source is called');
+
+	t.end();
+});
+
+// test('✈  fly.watch', function (t) {
```

Comment:
Looks like I overlooked this one. I remember the test never finished, which means I forgot a `yield` or a generator function inside the test. 

Fixing it now